### PR TITLE
Refacoring codebase to use module based dependency resolution

### PR DESCRIPTION
**Pull request by RaviPidaparthi**
Namespaces are complicating the external interface as well as the internal structure of the code. Moving to a module based dependency resolution will simplify the codebase.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,16 +1,20 @@
 {
   "name": "@speech/microsoft-speech-browser-sdk",
-  "version": "0.0.3",
+  "version": "0.0.1",
```
Comment (RaviPidaparthi) [inline package.json]: Resetting this so as to publish to npm freshly.

File: package.json [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
   "description": "Microsoft Speech SDK for browsers",
   "main": "distrib/speech.browser.sdk.js",
+  "types": "distrib/speech.browser.sdk.d.ts",
```
Comment (RaviPidaparthi) [inline package.json]: types is required to get the npm package working with other typescript projects.

File: samples/browser/Sample.html [hunk 3]
```diff
@@ -60,36 +63,43 @@ <h2 style="font-weight:500;">Microsoft Cognitive Services</h2>
         </tr>
     </table>
 
+    <!-- The SDK has a dependency on requirejs (http://requirejs.org/). -->
+    <script src="samples\require.js"></script>
```
Comment (RaviPidaparthi) [inline samples/browser/Sample.html]: RequireJs is now a dependency to use SDK.

File: package.json (hunk 1 above)
Comment (splusq) [inline package.json]: thanks

File: samples/browser/Sample.html (hunk 3 above)
Comment (splusq) [inline samples/browser/Sample.html]: that's fine but I am wondering if we should include it or reference a cdn?

Comment (RaviPidaparthi) [inline samples/browser/Sample.html]: We neednot include it in our package. The consumer of the sdk can choose how to reference it.

Comment (splusq) [inline samples/browser/Sample.html]: it's checked as of today - I am asking if we need to do that

**Pull request merged**